daemon/container_operations.go

```diff
@@ -967,10 +967,17 @@ func (daemon *Daemon) getNetworkedContainer(containerID, connectedContainerID st
 
 func (daemon *Daemon) releaseNetwork(container *container.Container) {
 	start := time.Now()
+	// If live-restore is enabled, the daemon cleans up dead containers when it starts up. In that case, the
+	// netController hasn't been initialized yet and so we can't proceed.
+	// TODO(aker): If we hit this case, the endpoint state won't be cleaned up (ie. no call to cleanOperationalData).
 	if daemon.netController == nil {
```

Comment:
Wondering if there's an actual case where `daemon.netController` _CAN_ be `nil` 🤔

I'm considering that that should (probably?) never happen, and if it _would_ we should probably make that a fatal error, and exit (before we even reach this code)

Reply:
Oh actually my comment isn't about an hypothetical situation:

* `(*Daemon).releaseNetwork` is called by `(*Daemon).Cleanup`: https://github.com/moby/moby/blob/670bc0a46c4ca03b75f1e72f73b8302bce90275a/daemon/start.go#L243
* Which is called by `(*Daemon).restore` on L449: https://github.com/moby/moby/blob/670bc0a46c4ca03b75f1e72f73b8302bce90275a/daemon/daemon.go#L449
* But `netController` is initialized by calling `(*Daemon).initNetworkingController` on L530: https://github.com/moby/moby/blob/670bc0a46c4ca03b75f1e72f73b8302bce90275a/daemon/daemon.go#L530